programs/stake/src/stake_instruction.rs

```diff
@@ -6128,4 +6217,265 @@ mod tests {
+            &stake_address,
+            &stake_account,
+            &vote_account,
+            &reference_vote_account,
+            Ok(()),
+        )[0];
+
+        assert_eq!(
+            stake_from(post_stake_account)
+                .unwrap()
+                .delegation
+                .deactivation_epoch,
+            current_epoch
+        );
+
+        // `reference_vote_account` has consistently voted and `vote_account` has not voted for the
+        // last `MINIMUM_DEACTIVATE_DELINQUENT_EPOCHS`. Try to deactivate an unrelated stake
+        // account.  Instruction will fail
+        let unrelated_vote_address = Pubkey::new_unique();
+        let unrelated_stake_address = Pubkey::new_unique();
+        let mut unrelated_stake_account = stake_account.clone();
+        unrelated_stake_account
+            .serialize_data(&StakeState::Stake(
+                Meta::default(),
+                new_stake(
+                    1, /* stake */
+                    &unrelated_vote_address,
+                    &VoteState::default(),
+                    1, /* activation_epoch */
+                    &stake_config::Config::default(),
+                ),
+            ))
+            .unwrap();
+
+        process_instruction_deactivate_delinquent(
+            &unrelated_stake_address,
+            &unrelated_stake_account,
+            &vote_account,
+            &reference_vote_account,
+            Err(StakeError::VoteAddressMismatch.into()),
+        );
+
+        // `reference_vote_account` has consistently voted and `vote_account` has voted once the
+        // last `MINIMUM_DEACTIVATE_DELINQUENT_EPOCHS` - 1.
+        // Instruction will succeed
+        let mut vote_state = VoteState::default();
+        vote_state.increment_credits(current_epoch - MINIMUM_DEACTIVATE_DELINQUENT_EPOCHS as Epoch);
+        vote_account
+            .serialize_data(&VoteStateVersions::new_current(vote_state))
+            .unwrap();
+        process_instruction_deactivate_delinquent(
+            &stake_address,
+            &stake_account,
+            &vote_account,
+            &reference_vote_account,
+            Ok(()),
+        );
+
+        // `reference_vote_account` has consistently voted and `vote_account` has voted once the
+        // last `MINIMUM_DEACTIVATE_DELINQUENT_EPOCHS`.
+        // Instruction will fail
+        let mut vote_state = VoteState::default();
+        vote_state
+            .increment_credits(current_epoch - (MINIMUM_DEACTIVATE_DELINQUENT_EPOCHS - 1) as Epoch);
+        vote_account
+            .serialize_data(&VoteStateVersions::new_current(vote_state))
+            .unwrap();
+        process_instruction_deactivate_delinquent(
+            &stake_address,
+            &stake_account,
+            &vote_account,
+            &reference_vote_account,
+            Err(StakeError::MinimumDeactivateDelinquentEpochsNotMet.into()),
+        );
+    }
```

Comment:
test for feature inactive fails?

Reply:
I thought about that, and found the excuse that we don't do this for any other gated stake instructions welcoming.